app/views/users/show.html.slim

```diff
@@ -33,10 +33,11 @@ h2 Staff details
   .row.pad-top-fifteen-px
     .columns.small-12
       = link_to "Change your password", edit_user_registration_path(@user)
--if current_user.elevated?
+-if current_user.elevated? && current_user != @user
```

Comment:
Here I'd put some sort of method that would describe the condition in an clear way. Something like:

`if user_has_the_permissions_and_not_themselves`